pkg/controller/core/workload_controller.go

```diff
@@ -826,9 +826,20 @@ func (r *WorkloadReconciler) admittedNotReadyWorkload(wl *kueue.Workload) (bool,
 	if podsReadyCond != nil && podsReadyCond.Status == metav1.ConditionTrue {
 		return false, 0
 	}
-	admittedCond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadAdmitted)
-	elapsedTime := r.clock.Since(admittedCond.LastTransitionTime.Time)
-	return true, max(r.waitForPodsReady.timeout-elapsedTime, 0)
+
+	if podsReadyCond == nil || podsReadyCond.Reason == kueue.WorkloadWaitForPodsStart {
```

Comment:
```suggestion
	if podsReadyCond == nil || podsReadyCond.Reason == kueue.WorkloadWaitForPodsStart || podsReadyCond.Reason == "PodsReady" {
```

IIUC, the `WaitForPodsStart` is compatible with the old `PodsReady` reason.

> We introduce new WorkloadWaitForPodsStart and WorkloadWaitForPodsRecovery reasons to distinguish the reasons of setting the PodsReady=false condition.

Reply:
We don't necessarily need to cover that case as we convert the old reason into a new one in the `generatePodsReadyCondition` function. However if you think it would be safer to take into account the old reason in both places I'm happy to update that

Reply:
It would be better to confirm both places (JobFramework and WorkloadController (here)) since they usually deploy kueue-external-manager to support non-upstream Jobs. In that case, the kueue-external-manager might be older than kueue-manger, which means kueue-external-manager JobFramework reconciler keeps repling on PodsReady reason.

It might be better to do following steps:

v0.11 (upcoming latest release): Check if podsReady reason is PodsReady in the workload-controller as well, which means verifying PodsReady with PodsReady reason case both in JobFramwork and WrokloadController. And add `[ACTION REQUIRED]` release note to upgrade JobFramwrork version in kueue-external-manager.
v0.12: Remove the redundant verification from WorkloadController, and then keep it only in JobFramework.

@mimowo Any thoughts?

Reply:
> This problem is more visible in case of external controllers, but is also present within build-in Kueue controllers, which is also why I'm thinking some integration test would be great.

SGTM, in that case, let's do 2 things in this PR:

1. Keep previous cases for PodsReady with `PodsReady` reason
2. Add Integration testings.